File: src/skyrimtemplates.pl
# Copyright 2015 Eli Foster

use warnings;
use strict;
use diagnostics;
use Switch;
use MediawikiAPI::API;
use SatanicBot::Utils;

my $ERRNO = $!;
my $mw = MediawikiAPI::API->new();
$mw->base_url('http://skyrim.gamepedia.com/api.php');
login();

print "Races/Skills/Cities/Houses? (0/1/2/3)\n";
my $input = <>;
chomp $input;
edit($input);

sub login {
    my @login = SatanicBot::Utils->get_secure_contents();
    $mw->login($login[0], $login[1]);
    return 1;
}

sub get_file {
    my ($type) = @_;
    switch ($type) {
        case 0 {
            open my $rfh, '<', 'info/races.txt' or die "Could not open info/races.txt $ERRNO\n";
            return <$rfh>;
        }
        case 1 {
            open my $sfh, '<', 'info/skills.txt' or die "Could not open info/skills.txt $ERRNO\n";
            return <$sfh>;
        }
        case 2 {
            open my $cfh, '<', 'info/cities.txt' or die "Could not open info/cities.txt $ERRNO\n";
            return <$cfh>;
        }
        case 3 {
            open my $hfh, '<', 'info/houses.txt' or die "Could not open info/houses.txt $ERRNO\n";
            return <$hfh>;
        }
    }
}

sub edit {
    my ($type) = @_;
    my @possible_types = (0, 1, 2, 3);
    my %types = (
        0 => 'Races',
        1 => 'Skills',
        2 => 'Cities',
        3 => 'Houses'
    );

    #while (my $line = get_file($type)) {
    #    chomp $line;
    my $line = 'User:TheSatanicSanta/Sandbox/BotTesting';
        my $edit_summary = 'Preparing to move Navboxes to more consistent and easy-to-find names.';
        #my $ref = $mw->get_page({title => $line});
        #my $text = $ref->{'*'};
        my $text;
        unless ($type == 2) {
            $text =~ s/\{\{$type\}\}/\{\{Navbox $types{$type}\}\}/g;
            $mw->edit_page($line, $text, $edit_summary);
        } else {
            $text =~ s/\{\{$type nav\}\}/\{\{Navbox $types{2}\}\}/g;
            $mw->edit_page($line, $text, $edit_summary);
        }

    #    print "Page \'$line\' has been edited.\n";
    #}
    #close $fh;
    #print 'File closed.';
    return 1;
}
exit 0;

File: src/ruby/skyrimtemplates.rb
require 'mediawiki_api'
require_relative 'wikiutils'
require_relative 'generalutils'

$mw = MediawikiApi::Client.new('http://skyrim.gamepedia.com/api.php')
$mw.log_in(General_Utils::File_Utils.get_secure(0).chomp, General_Utils::File_Utils.get_secure(1).chomp)
$other_mw = Wiki_Utils::Client.new('http://skyrim.gamepedia.com/api.php')

def get_file_lines(filename, nav_type)
  File.open(filename, 'r') do |file_handle|
    file_handle.each_line do |line|
      edit(line, nav_type)
    end
  end
end

def determine_file(nav_type)
  case nav_type
    when "Races"
      get_file_lines('../info/races.txt', nav_type)
    when "Skills"
      get_file_lines('../info/skills.txt', nav_type)
    when "Cities"
      get_file_lines('../info/cities.txt', nav_type)
    when "Houses"
      get_file_lines('../info/houses.txt', nav_type)
    else
      abort('What the hell?')
  end
end

def edit(page_name, nav_type)
  JSON.parse($other_mw.get_wikitext(page_name))["query"]["pages"].each do |revid, data |
    $revid = revid
    break
  end
  text = JSON.parse($other_mw.get_wikitext(page_name))["query"]["pages"][$revid]["revisions"][0]["*"]
  case nav_type
    when "Races"
      puts $other_mw.get_wikitext(page_name)
      text = text.gsub(/\{\{[Rr]aces\}\}/, "{{Navbox Races}}")
      $mw.edit(title: page_name, text: text)
      puts $other_mw.get_wikitext(page_name)
    when "Skills"
      text = text.gsub(/\{\{[Ss]kills\}\}/, "{{Navbox Skills}}")
    when "Cities"
      text = text.gsub(/\{\{[Cc]ity nav\}\}/, "{{Navbox Cities}}")
    when "Houses"
      text = text.gsub(/\{\{[Hh]ouses\}\}/, "{{Navbox Houses}}")
  end
end

puts "Which type of navbox would you like to change?\n"
nav_type = gets.chomp
edit('User:TheSatanicSanta/Sandbox/BotTesting', nav_type)
exit
